### DISCUSSION TITLE
API CHAT COMPLETIONS - ERROR [asyncio] Unclosed client session

### DISCUSSION THREAD

**Discussion by CodeGorger** [open-webui/open-webui#7251] [Issues]
# Bug Report

## Installation Method
Via the provided docker.

## Environment

- **Open WebUI Version:** 
0.3.35 and newer
- **Ollama (if applicable):**
0.3.8

## Expected Behavior:
No error, filter outlet gets called properly.

## Actual Behavior:

[Describe what actually happened.]

## Description

**Bug Summary:**
Error upon usage of endpoint  /api/chat/completions

## Reproduction Details

**Steps to Reproduce:**
Use the endpoint  /api/chat/completions of openwebui.

## Logs and Screenshots

**Docker Container Logs:**
INFO:     213.146.3.1:36791 - "POST /api/chat/completions HTTP/1.1" 200 OK
ERROR [asyncio] Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x7f75840e4a90>
ERROR [asyncio] Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x7f757effd6a0>, 26900.859)]']
connector: <aiohttp.connector.TCPConnector object at 0x7f7586e44a90>

## Additional Information

Using the endpoint /api/chat/completions always causes an error. What does work, is the inlet function of our filter. Then the response from the llm gets streamed properly. At the end of the streaming, the error triggers and the outlet never gets called anymore.
This problem does not happen, when using the openwebui frontend.
Assumption: The error happens, when the connection (after the streaming) doesn't properly shut down in "backend/open_webui/main.py".

**Comment by ay4t** [comment]
i have same issue when i using advanced RAG pipeline 
![2024-11-22-23-20-01](https://github.com/user-attachments/assets/09f45309-ecc3-4cdd-8f16-059b0c937d81)

but if i disable the outlet function it works very well but my pipeline functionality is not working
![image](https://github.com/user-attachments/assets/11745469-1278-4dc0-9375-a2a553a14b3b)

**Comment by arsaboo** [comment]
see #8074
